test/jsdom/encoding.js

```diff
@@ -0,0 +1,464 @@
+              res.end(`<!DOCTYPE html>
+              <html>
+                  <head>
+                      <meta charset="ISO-8859-5">
+                  </head>
+                  <body></body>
+              </html>`);
+              break;
+            case "/meta11":
+              res.writeHead(200, { "Content-Type": "text/plain;charset=ISO-8859-8" });
+              res.write(new Buffer([0xEF, 0xBB, 0xBF]));
+              res.end(`<!DOCTYPE html>
+              <html>
+                  <head>
+                      <meta charset="ISO-8859-5">
+                  </head>
+                  <body></body>
+              </html>`);
+              break;
+          }
+        });
+
+        server.listen(port, () => done());
+        testHost = "http://127.0.0.1:" + port;
+      });
+    });
+  });
+
+  after(() => {
+    return new Promise((done, rej) => {
+      server.close();
+      done();
+    });
+  });
+
+  specify("UTF-8", { async: true }, t => {
+    jsdom.env({
+      url: testHost + "/utf8",
+      encoding: "ISO-8859-1",
+      done(err, window) {
+        assert.ifError(err);
+
+        assert.equal(window.document.characterSet.toUpperCase(), "UTF-8", "document.characterSet");
+        assert.equal(window.document.body.textContent.trim(), "©");
+        assert.equal(window.testutf8, "©");
+        assert.equal(window.testiso88591, "©");
+
+        t.done();
+      },
+      features: {
+        FetchExternalResources: ["script"],
+        ProcessExternalResources: ["script"]
+      }
+    });
+  });
+
+  specify("ISO-8859-1", { async: true }, t => {
+    jsdom.env({
+      url: testHost + "/iso88591",
+      done(err, window) {
+        assert.ifError(err);
+
+        assert.equal(window.document.characterSet.toUpperCase(), "ISO-8859-1", "document.characterSet");
+        assert.equal(window.document.body.textContent.trim(), "©");
+        assert.equal(window.testutf8, "©");
+        assert.equal(window.testiso88591, "©");
+
+        t.done();
+      },
+      features: {
+        FetchExternalResources: ["script"],
+        ProcessExternalResources: ["script"]
+      }
+    });
+  });
+
+  specify("UTF-16BE", { async: true }, t => {
+    jsdom.env({
+      url: testHost + "/utf16be",
+      done(err, window) {
+        assert.ifError(err);
+
+        assert.equal(window.document.characterSet.toUpperCase(), "UTF-16BE", "document.characterSet");
+        assert.equal(window.document.body.textContent.trim(), "©");
+        assert.equal(window.testutf8, "©");
+        assert.equal(window.testiso88591, "©");
+
+        t.done();
+      },
+      features: {
+        FetchExternalResources: ["script"],
+        ProcessExternalResources: ["script"]
+      }
+    });
+  });
+
+  specify("UTF-16LE", { async: true }, t => {
+    jsdom.env({
+      url: testHost + "/utf16le",
+      done(err, window) {
+        assert.ifError(err);
+
+        assert.equal(window.document.characterSet.toUpperCase(), "UTF-16LE", "document.characterSet");
```

Comment:
I will try to normalize them thanks to this file : https://encoding.spec.whatwg.org/encodings.json